File: nodejs/tests/transport/spi_test.js
'use strict';

const { SPITransport } = require('../../packages/periph/src/transport/spi');

const SPI_BUS    = parseInt(process.env.SPI_BUS    || '0', 10);
const SPI_DEVICE = parseInt(process.env.SPI_DEVICE || '0', 10);
const SPI_SPEED  = parseInt(process.env.SPI_SPEED  || '1000000', 10);

let passed = 0;
let failed = 0;

function checkTrue(label, condition) {
    if (condition) { console.log('PASS', label); passed++; }
    else           { console.log('FAIL', label); failed++; }
}

const transport = new SPITransport(SPI_BUS, SPI_DEVICE, { maxSpeedHz: SPI_SPEED });

transport.write(Buffer.from([0x00]));
checkTrue('write accepted', true);

const readBuf = transport.read(1);
checkTrue('read returns 1 byte', readBuf.length === 1);

const wrBuf = transport.writeRead(Buffer.from([0x00]), 1);
checkTrue('writeRead returns 1 byte', wrBuf.length === 1);

transport.close();

console.log(`===DONE: ${passed} passed, ${failed} failed===`);
process.exit(failed === 0 ? 0 : 1);

File: python/tests/transport/spi_test.py
import _testconfig as cfg
from machine import SPI, Pin
from periph.transport.spi_micropython import SPITransport

passed = 0
failed = 0


def check_true(label, condition):
    global passed, failed
    if condition:
        print('PASS', label)
        passed += 1
    else:
        print('FAIL', label)
        failed += 1


spi = SPI(cfg.SPI_ID, baudrate=cfg.FREQ, sck=Pin(cfg.SCK), mosi=Pin(cfg.MOSI), miso=Pin(cfg.MISO))
cs  = Pin(cfg.CS, Pin.OUT, value=1)
transport = SPITransport(spi, cs)

transport.write(bytes([0x00]))
check_true('write accepted', True)

data = transport.read(1)
check_true('read returns 1 byte', len(data) == 1)

data = transport.write_read(bytes([0x00]), 1)
check_true('write_read returns 1 byte', len(data) == 1)

print('===DONE: {} passed, {} failed==='.format(passed, failed))
